datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -1610,6 +1629,126 @@ fn build_statistics_expr(
     Ok(statistics_expr)
 }
 
+fn build_like_match(
+    expr_builder: &mut PruningExpressionBuilder,
+) -> Option<Arc<dyn PhysicalExpr>> {
+    // column LIKE literal => (min, max) LIKE literal split at % => min <= split literal && split literal <= max
+    // column LIKE 'foo%' => min <= 'foo' && 'foo' <= max
+    // column LIKE '%foo' => min <= '' && '' <= max => true
+    // column LIKE '%foo%' => min <= '' && '' <= max => true
+    // column LIKE 'foo' => min <= 'foo' && 'foo' <= max
```

Comment:
in https://github.com/apache/datafusion/pull/12978#discussion_r1819023885 it was proposed to reword this and make a function doc comment.

If you want to keep as is, please make sure `=>` for implication and those comparisons can be easily read - maybe add braces?